utils/virtual_machines.py

```diff
@@ -52,11 +59,23 @@ def deploy_template(provider_key, vm_name, template_name=None, timeout=900, **de
     data = provider_crud.get_yaml_data()
 
     deploy_args.update(vm_name=vm_name)
+    if template_name is None:
+        try:
+            deploy_args.update(template=data['small_template'])
+        except KeyError:
+            raise ValueError('Provider {} does not have small_template defined in yaml').format(
+                provider_key)
+    else:
+        deploy_args.update(template=template_name)
     if isinstance(mgmt, RHEVMSystem):
         if 'default_cluster' not in deploy_args:
             deploy_args.update(cluster=data['default_cluster'])
     elif isinstance(mgmt, VMWareSystem):
-        if "allowed_datastores" not in deploy_args and "allowed_datastores" in data:
+        if 'resource_pool' in data["cap_and_util"] and purpose == 'capandu':
```

Comment:
This also means we can remove purpose. I think this is important because even as it is, purpose can be anything, and is not clearly defined, perhaps it would have been better to call it "capandu" and expect a boolean, but I prefer the other method above, unless I've missed something.